test/utils/Precompiles.t.sol

```diff
@@ -1,15 +1,20 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity >=0.8.22 <0.9.0;
 
+import { ISablierV2LockupDynamic } from "@sablier/v2-core/src/interfaces/ISablierV2LockupDynamic.sol";
+import { ISablierV2LockupLinear } from "@sablier/v2-core/src/interfaces/ISablierV2LockupLinear.sol";
+import { ISablierV2LockupTranched } from "@sablier/v2-core/src/interfaces/ISablierV2LockupTranched.sol";
+import { ISablierV2NFTDescriptor } from "@sablier/v2-core/src/interfaces/ISablierV2NFTDescriptor.sol";
+import { DeployOptimized as V2DeployOptimized } from "@sablier/v2-core/test/utils/DeployOptimized.sol";
 import { LibString } from "solady/src/utils/LibString.sol";
 
+import { Precompiles } from "../../precompiles/Precompiles.sol";
 import { ISablierV2Batch } from "../../src/interfaces/ISablierV2Batch.sol";
 import { ISablierV2MerkleLockupFactory } from "../../src/interfaces/ISablierV2MerkleLockupFactory.sol";
 
 import { Base_Test } from "../Base.t.sol";
-import { Precompiles } from "./Precompiles.sol";
 
-contract Precompiles_Test is Base_Test {
+contract Precompiles_Test is Base_Test, V2DeployOptimized {
```

Comment:
@andreivladbrg do you have any suggestion on how can I re-use `DeployOptimized` from core? Should I import it in `DeployOptimized` of periphery?

Reply:
I think importing it here would be alright - tho you can rename it to `V2CoreDeployOptimized`

Reply:
So I have removed the test `test_DeployProtocol` until we have a method to test it properly.

Reply:
hmm, that's weird, the `out-optimized` dir should contain the core contracts compiled as well since we import them

at any rate, the `deployProtocol() ` function calls the `deployCore()` and `deployPeriphery` functions which are individually tested, so it should be fine

this problem, and many more, are going to be fixed when address the "Package tethering" issue

Reply:
Yes you are correct. If no more changes required, can you please approve this?

Reply:
just brainstorming

should we use `deployProtocol` in `Base_Test` instead having to diferent functions:
- `deployPeripheryConditionally()`
- `deployDependencies()`

Reply:
hmm, good idea. 

Wouldn't we have to update precompiles during development every time we make changes to src contracts? `deployProtocol` uses precompiles while the current setup uses the build contracts.

Reply:
> Wouldn't we have to update precompiles during development every time we make changes to src contracts?

correct, so we can leave it as it is